gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,383 @@
+ * Runs through all the solutions currently available on the system, pulls the current
+ * setting for each supportedSetting and returns them in an object. Primary use case
+ * is for backing Capture tools that would allow a user to set up their GPII profile
+ * starting with the current settings for their applications on the local machine.
+ *
+ * @param {lifecycleManager.invokeSettingsHandlerGet} invokeSettingsHandlerGet - Invoker from the `lifecycleManager`
+ * necessary for extracting settings from a solution and it's spec.
+ * @param {lifecycleManager.variableResolver} resolver - Resolver for looking up handler specifications
+ * @param {lifecycleManager.userSession.localResolver} localResolver - Resolver for expanding configuration file paths
+ * from the solutions registry.
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {Object} options - Extra options for processing.
+ * @param {Array} options.solutionsList - If provided, only solutions in this list of `solutionsID`s will
+ * be captures. Example:
+ *
+ * '''json
+ * ["com.microsoft.windows.cursors", "com.freedomscientific.jaws"]
+ * '''
+ * @return {fluid.promise} Returns a promise resolving with the entire system settings capture.
+ */
+gpii.flowManager.captureSystemSettings = function (invokeSettingsHandlerGet, resolver, localResolver, solutions, options) {
+    var promiseTogo = fluid.promise();
+
+    var handlerGetPromises = [];
+
+    var safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
+        var promiseTogo = fluid.promise();
+        try {
+            invokeSettingsHandlerGet(solutionID, handlerSpec).then(function (data) {
+                // Fix up the nativeWindows and SPI Settings Handlers
+                var filteredData = gpii.flowManager.filterInvokeSettingsHandlersReturn(data, handlerSpec);
+                promiseTogo.resolve(filteredData);
+            }, function (err) {
+                promiseTogo.resolve({
+                    solutionId: solutionID,
+                    handlerSpec: handlerSpec,
+                    isError: true,
+                    message: "inner promise error",
+                    err: err
+                });
+            });
+        }
+        catch (err) {
+            promiseTogo.resolve({
+                solutionId: solutionID,
+                handlerSpec: handlerSpec,
+                isError: true,
+                message: "outer try error",
+                err: err
+            });
+        }
+        return promiseTogo;
+    };
+
+    var solutionsToFetch = solutions;
+    if (options.solutionsList) {
+        solutionsToFetch = {};
+        fluid.each(solutions, function (solution, solutionID) {
+            if (options.solutionsList.indexOf(solutionID) >= 0) {
+                solutionsToFetch[solutionID] = solution;
+            }
+        });
+    }
+
+    fluid.each(solutionsToFetch, function (solution, solutionID) {
```

Comment:
This part should be broken into its own function that accepts a list of solutions and return a promise whose resolved value is settings read from the system.

Reply:
I split these up into in to their own functions.